### PR TITLE
[pytorch] remove unused flags from code analyzer & move format support to python

### PR DESCRIPTION
**Pull request by ljk53**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #37404 [pytorch] add '__BASE__' section to op deps to factor out frequently used util ops
* **#37393 [pytorch] remove unused flags from code analyzer & move format support to python**

Summary:
Simplify the code analyzer by removing some unused flags and moving the
different format printer logic to python script. It's easier to add other
post processing logic to adapt to different BUCK build configs.

Differential Revision: [D21280836](https://our.internmc.facebook.com/intern/diff/D21280836)

### CODE REVIEW

**Review by iseeyuan — APPROVED** [review]
LGTM

File: tools/code_analyzer/op_deps_processor.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    "${op_name}" -> "${dep_name}";
+""")
+
+
+def load_op_deps(fname):
+    with open(fname, 'r') as stream:
+        return yaml.safe_load(stream)
+
+
+def convert(fname, graph, OUTPUT, OP, OP_DEP):
```
Comment (bhosmer) [inline tools/code_analyzer/op_deps_processor.py:57]: nit: might be slightly easier to distinguish between these params and the template globals if the params were lowercase

**Review by bhosmer — APPROVED** [review]

File: tools/code_analyzer/op_deps_processor.py (hunk 1 above)
Comment (ljk53) [reply tools/code_analyzer/op_deps_processor.py:57]: hmm, on the other hand I want to distinguish between code template v.s. code after substitution; maybe I can use "_template" suffix instead...

Comment (bhosmer) [reply tools/code_analyzer/op_deps_processor.py:57]: ooo yeah, `_template` is perfect

**Pull request closed without merging**